File: src/test/scala/com/github/kayjamlang/tests/expressions/access/CompanionAccessExpressionExpressionTest.scala
package com.github.kayjamlang.tests.expressions.access

import com.github.kayjamlang.core.KayJamLexer
import com.github.kayjamlang.core.KayJamParser
import com.github.kayjamlang.core.expressions.CompanionAccessExpression
import com.github.kayjamlang.core.expressions.Expression
import com.github.kayjamlang.core.expressions.VariableLinkExpression
import org.junit.BeforeClass
import org.junit.Test
import org.junit.Assert._


object CompanionAccessExpressionExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("Test::test"))
  }
}

class CompanionAccessExpressionExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = CompanionAccessExpressionExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[CompanionAccessExpression], expression getClass)
    val companionAccessExpression = expression.asInstanceOf[CompanionAccessExpression]
    assertEquals("Test", companionAccessExpression.className)
    assertSame(classOf[VariableLinkExpression], companionAccessExpression.child getClass)
    val variableLinkExpression = companionAccessExpression.child.asInstanceOf[VariableLinkExpression]
    assertEquals("test", variableLinkExpression.name)
  }
}

File: src/test/java/com/github/kayjamlang/tests/expressions/access/CompanionAccessExpressionExpressionTest.java
package com.github.kayjamlang.tests.expressions.access;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.CompanionAccessExpression;
import com.github.kayjamlang.core.expressions.VariableLinkExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class CompanionAccessExpressionExpressionTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("Test::test"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(CompanionAccessExpression.class, expression.getClass());

        CompanionAccessExpression companionAccessExpression = (CompanionAccessExpression) expression;
        assertEquals("Test", companionAccessExpression.className);
        assertSame(VariableLinkExpression.class, companionAccessExpression.child.getClass());

        VariableLinkExpression variableLinkExpression = (VariableLinkExpression) companionAccessExpression.child;
        assertEquals("test", variableLinkExpression.name);
    }
}